### PR TITLE
Further concurrency

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: sn_node/src/node/flow_ctrl/mod.rs [hunk 1]
```diff
@@ -90,7 +90,7 @@ impl FlowCtrl {
 
         // Our channel to process _all_ cmds. If it can, they are processed off thread with latest context,
         // otherwise they are sent to the blocking process channel
-        let (flow_ctrl_cmd_sender, flow_ctrl_cmd_reciever) = mpsc::channel(STANDARD_CHANNEL_SIZE);
```
Comment (RolandSherwin) [inline sn_node/src/node/flow_ctrl/mod.rs]: This commit is redundant as the change is being reverted back in the following [commit](https://github.com/maidsafe/safe_network/pull/2205/commits/eedb19b1b95df2d5de7731e328828fb0c6d2f908).

File: sn_node/src/node/flow_ctrl/mod.rs [hunk 2]
```diff
@@ -121,7 +121,7 @@ impl FlowCtrl {
         };
 
         // first start listening for msgs
-        let cmd_channel_for_msgs = blocking_cmd_sender_channel.clone();
```
Comment (RolandSherwin) [inline sn_node/src/node/flow_ctrl/mod.rs:124]: A very small one but we can probably save a "clone" here by directly cloning at the function call.

File: sn_node/src/node/messaging/mod.rs [hunk 3]
```diff
@@ -75,14 +75,28 @@ impl MyNode {
...
+        let sender_addr = sender.addr();
         // alternatively we could flag in msg kind for this...
         // todo: this sender is actually client + forwarder ip....
 
-        // we've forwaded it to ourselves as we're the holder. This prevents a loop.
-        // TODO: cut that wee loop down
-        let is_from_us = sender.addr() == context.info.addr;
+        // It's been forwarded, by another node, or ourself. This prevents repeatdly
+        // forwarding client msgs
+        let is_from_elder_node = !context
```
Comment (oetyng) [inline sn_node/src/node/messaging/mod.rs:84]: I don't get this.
We say that it is from an elder node if none of our members sent it?

Comment (oetyng) [reply sn_node/src/node/messaging/mod.rs:84]: I see this further down
`debug!("{msg_id:?} is from another network node?`

So, is the variable meant to say that it is a msg from another section?

Comment (joshuef) [reply sn_node/src/node/messaging/mod.rs]: Aye, this makes no sense atm. I thought I'd sorted something in another branch with this but pulled it in, but it's wrong wrong wrong. 

Removed this change altogether 👍 🙇

**Review by RolandSherwin — APPROVED** [review]

**Review by RolandSherwin — APPROVED** [review]

**Pull request merged**